dvc/stage/exceptions.py

```diff
@@ -44,6 +44,11 @@ class StageFileBadNameError(DvcException):
     pass
 
 
+class StageFileIgnoredError(StageFileBadNameError):
+    def __init__(self, path):
+        super().__init__(f"bad DVC file name '{path}' is git-ignored.")
+
+
```

Comment:
We already have `FileIsGitIgnored`, do we need this at all?

Reply:
I added this since `FileIsGitIgnored` is specific to `dvc/dvcfile.py`, and there is other code elsewhere where we specifically catch & handle `StageFileBadNameError`

Reply:
Hmm, maybe I am missing something here, but I could only find this during exception use during the checkout (where this does not matter) and during `dump` and `create_stage` where we never catch them.

Btw, `create_stage` might not be a good place to check them, we can just check this during the `dump`.

Reply:
I'm not sure that moving the check to `dump` works for us, since the `dump` happens last in add/move (after we've already copied/moved objects into cache and linked them back into the workspace). If we are going to fail because the destination `.dvc`/`dvc.yaml`/`dvc.lock` file would be in an invalid location, that failure should happen before we actually do any other processing on the output itself.

Reply:
changed to use `FileIsGitIgnored` in f9f45e2